spec/services/live_stream_updater_spec.rb

```diff
@@ -0,0 +1,52 @@
+require "rails_helper"
+require "gds_api/test_helpers/publishing_api"
+
+RSpec.describe LiveStreamUpdater do
+  include GdsApi::TestHelpers::PublishingApi
+
+  #LiveStream objects have a default state of false when created.
+  let(:live_stream) { LiveStream.first_or_create }
+  let(:state_params_on) { true }
+  let(:state_params_off) { false }
+  let(:content_item_live_stream_off) { File.read(Rails.root.join + "spec/fixtures/coronavirus_content_item.json") }
+
+  context "Succesful interaction with publishing api" do
+
+    let(:updater) { described_class.new( live_stream, state_params_on ) }
+
+    before :each do
+      stub_any_publishing_api_put_content
+      stub_any_publishing_api_publish
+      stub_publishing_api_has_item(JSON.parse(content_item_live_stream_off))
+    end
+
+    it "#updated?" do
+      expect(updater.updated?).to be true
+      expect(LiveStream.last.state).to be true
+    end
+
+    it "#published?" do
+      expect(updater.published?).to be true
+      expect(LiveStream.last.state).to be true
+    end
+  end
+
+  context "Unsuccesful interaction with publishing api" do
+
+    let(:updater) { described_class.new( live_stream, state_params_on ) }
+
+    before :each do
+      stub_any_publishing_api_call_to_return_not_found
```

Comment:
This should raise a `GdsApi::HTTPNotFound` error which I don't think we're catching yet. The full list is here: https://github.com/alphagov/gds-api-adapters/blob/bd031098cbfcee4934ad0648a9b9e29793e6f776/lib/gds_api/exceptions.rb#L62-L93